java/org/apache/catalina/startup/ContextConfig.java

```diff
@@ -122,6 +126,11 @@
 
     private static final Log log = LogFactory.getLog(ContextConfig.class);
 
+    private static final int DEFAULT_CLASS_CACHE_SIZE = 16384;
+
+    private static final float DEFAULT_LOAD_FACTOR = .75f;
+
+    private static int CONCURRENCY_LEVEL = 1;
```

Comment:
Ok for the map size, maybe (it seems very large, 16k classes ?), I don't quite see the point of the rest. The concurrency may be up to the utility executor thread count, and I don't think this is an optimization which makes any difference.

Reply:
Resolved with latest commit.